modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep_chunks/aah_sweep_chunk.cc

```diff
@@ -39,277 +37,242 @@ AAH_SweepChunk::Sweep(AngleSet& angle_set)
-        else if (not boundary)
+        else if (not is_boundary_face)
           ++preloc_face_counter;
 
-        sweep_dependency_interface_.SetupIncomingFace(
-          f, cell_mapping_->NumFaceNodes(f), face.neighbor_id_, local, boundary);
-
-        aah_sweep_depinterf.in_face_counter = in_face_counter;
-        aah_sweep_depinterf.preloc_face_counter = preloc_face_counter;
-
         // IntSf_mu_psi_Mij_dA
-        const size_t cf = sweep_dependency_interface_.current_face_idx_;
-        const auto& M_surf_f = (*M_surf_)[cf];
-        const double mu = face_mu_values_[cf];
-        const size_t num_face_nodes = sweep_dependency_interface_.num_face_nodes_;
+        const size_t num_face_nodes = cell_mapping.NumFaceNodes(f);
         for (int fi = 0; fi < num_face_nodes; ++fi)
         {
-          const int i = cell_mapping_->MapFaceNode(cf, fi);
+          const int i = cell_mapping.MapFaceNode(f, fi);
+
           for (int fj = 0; fj < num_face_nodes; ++fj)
           {
-            const int j = cell_mapping_->MapFaceNode(cf, fj);
-            const double* psi = sweep_dependency_interface_.GetUpwindPsi(fj);
-            const double mu_Nij = -mu * M_surf_f[i][j];
-            Amat_[i][j] += mu_Nij;
-
-            if (psi == nullptr)
+            const int j = cell_mapping.MapFaceNode(f, fj);
+
+            const double mu_Nij = -face_mu_values[f] * M_surf[f][i][j];
+            Amat[i][j] += mu_Nij;
+
+            const double* psi;
+            if (is_local_face)
+              psi = fluds.UpwindPsi(spls_index, in_face_counter, fj, 0, as_ss_idx);
+            else if (not is_boundary_face)
+              psi = fluds.NLUpwindPsi(preloc_face_counter, fj, 0, as_ss_idx);
+            else
+              psi = angle_set.PsiBndry(cell_face.neighbor_id_,
+                                       direction_num,
+                                       cell_local_id,
+                                       f,
+                                       fj,
+                                       gs_gi,
+                                       gs_ss_begin,
+                                       IsSurfaceSourceActive());
+
+            if (not psi)
               continue;
 
-            for (int gsg = 0; gsg < gs_ss_size_; ++gsg)
-              b_[gsg][i] += psi[gsg] * mu_Nij;
+            for (int gsg = 0; gsg < gs_ss_size; ++gsg)
+              b[gsg][i] += psi[gsg] * mu_Nij;
           } // for face node j
         }   // for face node i
       }     // for f
 
       // Looping over groups, assembling mass terms
-      for (int gsg = 0; gsg < gs_ss_size_; ++gsg)
+      for (int gsg = 0; gsg < gs_ss_size; ++gsg)
       {
-        g_ = gs_gi_ + gsg;
-        gsg_ = gsg;
-        sigma_tg_ = sigma_t[g_];
-
-        const auto& M = *M_;
-        const auto& m2d_op = groupset_.quadrature_->GetMomentToDiscreteOperator();
+        double sigma_tg = sigma_t[gs_gi + gsg];
 
         // Contribute source moments q = M_n^T * q_moms
```

Comment:
what is that transpose? I bet m2d and d2m are built with the same meaning for rows and cols while, in reality they should not be that way. If I am right, this can be really confusing for someone reading the code after reading a paper. Let's remember this when you tackle the angular qaudrature.